lci_strategic_plugin/src/lci_strategic_plugin.cpp

```diff
@@ -952,22 +825,14 @@ void LCIStrategicPlugin::planWhenAPPROACHING(const cav_srvs::PlanManeuversReques
     ts_params.case_num = CASE_8;
     print_params(ts_params);
   }
+  else if (distance_remaining_to_traffic_light <= desired_distance_to_stop && in_tdb) // Given ET is in TDB, but vehicle is too close to intersection
+  {
+    ROS_DEBUG_STREAM("ET is still in TDB despite the vehicle being in desired distance to start stopping. Trying to handle this edge case gracefully...");
```

Comment:
I guess you don't need to add handlestopping again? Would it be taken care of by the following logics and checks?

Reply:
I also have a similar question as @Saeid-Solei . won't you still face this condition if the vehicle is in the middle of the stopping process?
Also, we have "if else", does the condition have a final "else"? Because I see more "if" statements later

Reply:
Yep. Next few logics directly start checking whether if the vehicle should cruise and stop, directly stop, or handleFailureCase-s when not within safety stopping distance

Reply:
sounds good. In that case add something to that log statement saying condition handled by later logic or something.